src/main/java/org/aion/wallet/account/AccountManager.java

```diff
@@ -0,0 +1,277 @@
+            byte[] seed = new SeedCalculator().calculateSeed(mnemonic, DEFAULT_MNEMONIC_SALT);
+            final ECKey key = new SeededECKeyEd25519(seed);
+            final byte[] fileContent = keystoreFormat.toKeystore(key, password);
+            return getUnlockedAccountFromKey(key, fileContent, password, shouldKeep);
+        } catch (final Exception e) {
+            throw new ValidationException(e);
+        }
+    }
+
+    private AccountDTO getUnlockedAccountFromKey(final ECKey key, final byte[] fileContent, final String password, final boolean shouldKeep) throws UnsupportedEncodingException, ValidationException {
+        String address = TypeConverter.toJsonHex(KeystoreItem.parse(fileContent).getAddress());
+        final AccountDTO accountDTO;
+        if (shouldKeep) {
+            if (!Keystore.exist(address)) {
+                address = Keystore.create(password, key);
+                if (!address.equals("0x")) {
+                    accountDTO = createAccountWithPrivateKey(address, key.getPrivKeyBytes());
+                } else {
+                    throw new ValidationException("Failed to save keystore file");
+                }
+            } else {
+                throw new ValidationException("Account already exists!");
+            }
+        } else {
+            if (!addressToAccount.keySet().contains(address)) {
+                accountDTO = createAccountWithPrivateKey(address, key.getPrivKeyBytes());
+            } else {
+                throw new ValidationException("Account already exists!");
+            }
+        }
+        processAccountAdded(accountDTO, fileContent, false);
+        return accountDTO;
+    }
+
+    public SortedSet<TransactionDTO> getTransactions(final String address) {
+        return addressToTransactions.getOrDefault(address, Collections.emptySortedSet());
+    }
+
+    public TxInfo getLastTxInfo(final String address) {
+        return addressToLastTxInfo.get(address);
+    }
+
+    public void updateTxInfo(final String address, final TxInfo txInfo) {
+        addressToLastTxInfo.put(address, txInfo);
+    }
+
+    public List<AccountDTO> getAccounts() {
+        for (Map.Entry<String, AccountDTO> entry : addressToAccount.entrySet()) {
+            AccountDTO account = entry.getValue();
+            account.setBalance(BalanceUtils.formatBalance(balanceProvider.apply(account.getPublicAddress())));
+            entry.setValue(account);
+        }
+        return new ArrayList<>(addressToAccount.values());
+    }
+
+    public Set<String> getAddresses() {
+        return addressToAccount.keySet();
+    }
+
+    public AccountDTO getAccount(final String publicAddress) {
+        final String name = getStoredAccountName(publicAddress);
+        final String balance = BalanceUtils.formatBalance(balanceProvider.apply(publicAddress));
+        return new AccountDTO(name, publicAddress, balance, currencySupplier.get());
+    }
+
+    public void updateAccount(final AccountDTO account) {
```

Comment:
updateAccountName maybe?

Reply:
I was thinking that the method names should describe generic operations and the actual nature of the update would not be visible from outside, allowing us to hide the implementation's details